depthai_nodes/parser_generator.py

```diff
@@ -53,7 +54,22 @@ def build(self, nn_archive: dai.NNArchive, head_index: int = None) -> Dict:
                 )
 
             head = decode_head(head)
-            parsers[index] = pipeline.create(parser).build(head)
+            sig = inspect.signature(parser.build)
+            if "inputs_size" in sig.parameters:
+                inputs_size = []
+                for input in nn_archive.getConfig().model.inputs:
+                    if input.layout == "NHWC":
+                        _, height, width, _ = input.shape
+                    elif input.layout == "NCHW":
+                        _, _, height, width = input.shape
+                    else:
+                        raise ValueError(
```

Comment:
We shouldn't raise an error because some models will have input layout different from NCHW/NHWC. E.g. Gaze estimation model requires 3rd input to be of shape [1,3] (there are also some embedding models and LLM in the future maybe). What we can do is to set the `input_size` to the whole input shape (e.g. [1,3]) and print a warning saying this input size is used for input `<x>`. Thoughts?

Reply:
Good point. I agree with passing the whole input_shape and then the parser is responsible for getting input size (height, width) for it since specific parser "knows" what type of inputs it expects (e.g. YuNet expects image input so we can get height and width inside its build()). In case layout is NHWC would it make sense to convert to NCHW here so we always work with NCHW internally?

Reply:
> We shouldn't raise an error because some models will have input layout different from NCHW/NHWC.

We only need to set the `inputs_size` parameter for two models for now (YuNet and LineDetection). So this part of the code wouldn't be run for GazeEstimation. However, I see how this is a bit buggy and can cause trouble later on. Will move this part to parser as suggested by @klemen1999.

Reply:
Yes, let's put this into build() and set `self.input_shape` automatically. And migh be good to check if layout is NHWC to transpose it to NCHW so it is easier to process internally